plugins/bundle/plugin.go

```diff
@@ -429,3 +504,74 @@ func (p *Plugin) configDelta(newConfig *Config) (map[string]*Source, map[string]
 
 	return newBundles, updatedBundles, deletedBundles
 }
+
+func saveBundleToDisk(path string, name string, b *bundle.Bundle) error {
+
+	bundleDir := filepath.Join(path, name)
+
+	if _, err := os.Stat(filepath.Join(bundleDir, "bundle.tar.gz")); err == nil {
+		err := saveCurrentBundleToDisk(bundleDir, ".bundle.tar.gz.tmp", b)
+		if err != nil {
+			return os.Remove(filepath.Join(bundleDir, ".bundle.tar.gz.tmp"))
+		}
+	} else {
+		err := saveCurrentBundleToDisk(bundleDir, ".bundle.tar.gz.tmp", b)
+		if err != nil {
+			os.Remove(filepath.Join(bundleDir, ".bundle.tar.gz.tmp"))
+			return err
+		}
+	}
```

Comment:
There's one difference: If you try to save a bundle to disk when there isn't one already, we return the error that occurred while saving it. OTOH, if there is an exiting bundle and the new one could not be saved, we return the error from removing the temp file.

Reply:
Ahhh ok, I see it now. Looking at where the error is used, would we want the bundle status to go into an error state if the temp file cleanup failed? Seems like maybe a log message or something to warn about it, but otherwise the bundle that was activated is still ok, right?

I guess thinking about it more we should be careful to at least log any of the errors (from save, cleanup, etc), especially if they don't affect the bundle status. Future us reading some bug report with a log will probably be thankful.

Reply:
Added a warning log to capture errors that don't affect bundle status.